### PR TITLE
Nvector solution

### PR DESCRIPTION
**Pull request by mikaelar**

### CODE REVIEW

File: Sem. 07/Practicum/Nvector.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+};
+const Nvector operator+(const Nvector&, const Nvector&);
+const Nvector operator-(const Nvector&, const Nvector&);
+const Nvector operator*(const Nvector&, int);
+
+bool operator||(const Nvector&, const Nvector&);
+bool operator||(const Nvector&, const Nvector&);
+bool areOrt(const Nvector&, const Nvector&);
+double length(const Nvector&);
```
Comment (GeorgiTerziev02) [inline Sem. 07/Practicum/Nvector.h:43]: It is better to move this inside the Nvector class

File: Sem. 07/Practicum/Nvector.cpp [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+double length(const Nvector& obj) {
+	int sum = 0;
+	for (int i = 0; i < obj.getSize(); i++) {
+		sum += obj.getComponents(i) * obj.getComponents(i);
+	}
+	return sqrt(sum);
+}
+
+void print() {
```
Comment (GeorgiTerziev02) [inline Sem. 07/Practicum/Nvector.cpp:208]: I don't think this is a good name for this function

File: Sem. 07/Practicum/Nvector.cpp [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+		int num = first.getComponents(i) * second.getComponents(i);
+		sum += num;
+	}
+
+	if (sum == 0)
+	{
+		return true;
+	}
+
+	return false;
```
Comment (GeorgiTerziev02) [inline Sem. 07/Practicum/Nvector.cpp:197]: Use the shorter syntax - return sum == 0

File: Sem. 07/Practicum/Nvector.cpp [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+		components[i] = 0;
+	}
+}
+
+int Nvector::getSize() const
+{
+	return size;
+}
+
+int Nvector::getComponents(int index) const
```
Comment (GeorgiTerziev02) [inline Sem. 07/Practicum/Nvector.cpp:19]: this does the same job as operator []

File: Sem. 07/Practicum/Nvector.cpp [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+{
+	if (first.getSize() != second.getSize()) {
+		print();
+		return false;
+	}
+
+	int sum = 0;
+
+	for (int i = 0; i < first.getSize(); i++) {
+		int num = first.getComponents(i) * second.getComponents(i);
```
Comment (GeorgiTerziev02) [inline Sem. 07/Practicum/Nvector.cpp:188]: No need of this temp variable

File: Sem. 07/Practicum/Nvector.cpp [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+	return a;
+}
+
+Nvector twoDimCoordSystem::getB() const {
+	return b;
+}
+
+Nvector twoDimCoordSystem::getVectorByCoordinates(int k, int m)
+{
+	Nvector result;
```
Comment (GeorgiTerziev02) [inline Sem. 07/Practicum/Nvector.cpp:228]: Nvector result(2)

File: Sem. 07/Practicum/Nvector.cpp [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+
+	return result;
+}
+
+Nvector twoDimCoordSystem::getCoordinatesByVector(int l, int n)
+{
+	Nvector coord;
+
+	coord[1] = (a[0] * n - l * a[1]) / (b[1] * a[0] - b[0] * a[1]);
+	coord[0] = (l - coord[1] * b[0]) / a[0];
```
Comment (GeorgiTerziev02) [inline Sem. 07/Practicum/Nvector.cpp:241]: Not sure if these calculations are correct, please double-check them

**Review by YavorAlexandrov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**